File: crawler/src/entity/album.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
    const Album = sequelize.define('Album', {
        id: {
            type: DataTypes.BIGINT,
            primaryKey: true,
            autoIncrement: true,
        },
        name: {
            type: DataTypes.STRING,
            allowNull: false,
        },
        cover: {
            type: DataTypes.STRING,
            allowNull: false,

        },
        releaseDate: {
            type: DataTypes.DATE,
            allowNull: true,
            field: 'release_date',
        },
        musicBrainzId: {
            type: DataTypes.STRING,
            allowNull: true,
            field: 'music_brainz_id',
        },
        musicBrainzName: {
            type: DataTypes.STRING,
            allowNull: true,
            field: 'music_brainz_name',
        },
        messageId: {
            type: DataTypes.INTEGER,
            allowNull: false,
            field: 'message_id',
        },
        genres: {
            type: DataTypes.JSONB,
            allowNull: false,
            defaultValue: [],
        },
    }, {
        tableName: 'album',
        timestamps: true,
        underscored: true,
        createdAt: 'created_at',
        updatedAt: 'updated_at',
    });

    Album.associate = (entity) => {
        Album.belongsTo(entity.Artist, {
            foreignKey: 'artist_id',
            as: 'artist',
        });
        Album.hasMany(entity.Track, {
            foreignKey: 'album_id',
            as: 'tracks',
        });
    };

    return Album;
};

File: api/src/main/java/com/citytube/api/models/Album.java
package com.citytube.api.models;



import io.hypersistence.utils.hibernate.type.json.JsonBinaryType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.Type;

import java.time.LocalDate;
import java.util.List;


@EqualsAndHashCode(callSuper = true)
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "album")
public class Album extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    private String cover;

    private LocalDate releaseDate;

    private String musicBrainzId;

    private String musicBrainzName;

    @Column(nullable = false)
    private Integer messageId;

    @Type(JsonBinaryType.class)
    @Column(columnDefinition = "jsonb")
    private List<String> genres;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "artist_id", nullable = false)
    private Artist artist;

    @OneToMany(mappedBy = "album", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Track> tracks;


}